wp-includes/js/service-worker-navigation-routing.js

```diff
@@ -0,0 +1,102 @@
+/* global console, ERROR_OFFLINE_URL, ERROR_500_URL, SHOULD_STREAM_RESPONSE, STREAM_HEADER_FRAGMENT_URL, ERROR_500_BODY_FRAGMENT_URL, ERROR_OFFLINE_BODY_FRAGMENT_URL, STREAM_HEADER_FRAGMENT_QUERY_VAR, BLACKLIST_PATTERNS */
+
+const isStreamingResponses = SHOULD_STREAM_RESPONSE && wp.serviceWorker.streams.isSupported();
+
+wp.serviceWorker.routing.registerRoute( new wp.serviceWorker.routing.NavigationRoute(
+	async function ( { event } ) {
+		const { url } = event.request;
+
+		const handleResponse = ( response ) => {
+			if ( response.status < 500 ) {
+				if ( response.redirected ) {
+					const redirectedUrl = new URL( response.url );
+					redirectedUrl.searchParams.delete( STREAM_HEADER_FRAGMENT_QUERY_VAR );
+					const script = `<script>history.replaceState( {}, '', ${ JSON.stringify( redirectedUrl.toString() ) } );</script>`;
+					return response.text().then( ( body ) => {
+						return new Response( script + body );
+					} );
+				} else {
+					return response;
+				}
+			}
+			const channel = new BroadcastChannel( 'wordpress-server-errors' );
```

Comment:
Not a blocker at this point, but it would be needed to define the scenario for browsers that do not support BroadcastChannel (https://caniuse.com/#search=broadcastchannel)

Reply:
Yes, this is captured in #81.